sae_lens/training/activations_store.py

```diff
@@ -374,15 +380,23 @@ def _iterate_tokenized_sequences(self) -> Generator[torch.Tensor, None, None]:
         else:
             tokenizer = getattr(self.model, "tokenizer", None)
             bos_token_id = None if tokenizer is None else tokenizer.bos_token_id
-            yield from concat_and_batch_sequences(
-                tokens_iterator=self._iterate_raw_dataset_tokens(),
-                context_size=self.context_size,
-                begin_batch_token_id=(bos_token_id if self.prepend_bos else None),
-                begin_sequence_token_id=None,
-                sequence_separator_token_id=(
-                    bos_token_id if self.prepend_bos else None
-                ),
-            )
+
+            if self.disable_concat_sequences:
+                for tokens in self._iterate_raw_dataset_tokens():
+                    if len(tokens) >= self.context_size:
+                        yield tokens[: self.context_size]
+            else:
+                sequence_separator_token_id = None
+                if self.prepend_bos and not self.exclude_bos_between_sequences:
+                    sequence_separator_token_id = bos_token_id
+
+                yield from concat_and_batch_sequences(
```

Comment:
It would be better to adapt the `concat_and_batch_sequences` function to handle `disable_concat_sequences`, so then the pretokenize runner could also gain this capability too.